pkg/stanza/fileconsumer/trie.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TRIE data structure inspired by https://github.com/dghubble/trie
+// This differs from the original trie.
+
+package fileconsumer
+
+type Trie struct {
```

Comment:
I think we need thorough tests for the trie itself. What do you think about adding the trie and dedicated tests to `fileconsumer/internal` in a separate PR?

Reply:
Yeah, we can do that. Will reduce the the time complexity of this PR